tink-envelope-encryption-sample/encrypted_keyset_cli.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"log"
+	"os"
+
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/core/registry"
+	"github.com/google/tink/go/integration/gcpkms"
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/tink"
+	"google.golang.org/api/option"
+)
+
+var (
+	mode           = flag.String("mode", "", "The operation to perform: generate, encrypt, or decrypt")
+	keysetPath     = flag.String("keyset_path", "", "Path to the keyset used for encryption")
+	kekURI         = flag.String("kek_uri", "", "The Cloud KMS URI of the key encryption key")
+	gcpCredential  = flag.String("gcp_credential_path", "", "Path to the GCP credentials JSON file")
+	inputPath      = flag.String("input_path", "", "Path to the input file")
+	outputPath     = flag.String("output_path", "", "Path to the output file")
+	associatedData = flag.String("associated_data", "", "Optional associated data to use with the encryption operation")
+)
+
+func main() {
+	flag.Parse()
+
+	// load master key
+	ctx := context.Background()
+	gcpClient, err := gcpkms.NewClientWithOptions(ctx, *kekURI, option.WithCredentialsFile(*gcpCredential))
+	if err != nil {
+		log.Fatal(err)
+	}
+	registry.RegisterKMSClient(gcpClient)
+
+	masterKey, err := gcpClient.GetAEAD(*kekURI)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	switch *mode {
+	case "generate":
+		if err := generateKeyset(*outputPath, masterKey); err != nil {
+			log.Fatalf("Error generating keyset: %v", err)
+		}
+	case "encrypt", "decrypt":
+		if err := processFile(*mode, *outputPath, *inputPath, *keysetPath, *associatedData, masterKey); err != nil {
+			log.Fatalf("Error: %v", err)
+		}
+	default:
+		log.Fatalf("Unsupported mode %s. Please choose 'generate', 'encrypt', or 'decrypt'", *mode)
+	}
+}
+
+func generateKeyset(outputPath string, masterKey tink.AEAD) error {
+
+	var err error
+	// create output file
+	_, err = os.Stat(outputPath)
+	if err == nil {
+		log.Fatal(errors.New("output file must not exist"))
+	}
+	f, err := os.Create(outputPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	// generate a new key
+	keyTemplate := aead.AES128GCMKeyTemplate()
```

Comment:
```suggestion
	keyTemplate := aead.AES256GCMKeyTemplate()
```
Let's use a 256 key, which is also what the default KMS symmetric encryption algorithm uses behind the scenes (so that for the migration CUJ, this doesn't end up being a key downgrade)

Reply:
Good catch!